# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLShader.cpp
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGLShader.cpp
--- a/dom/canvas/WebGLShader.cpp	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGLShader.cpp	Wed Jan 08 22:19:14 2020 +0000
@@ -148,27 +148,24 @@
   CopyASCIItoUTF16(mCompilationLog, *out);
 }
 
-JS::Value WebGLShader::GetShaderParameter(GLenum pname) const {
+MaybeWebGLVariant WebGLShader::GetShaderParameter(GLenum pname) const {
   switch (pname) {
     case LOCAL_GL_SHADER_TYPE:
-      return JS::NumberValue(mType);
+      return AsSomeVariant(mType);
 
     case LOCAL_GL_DELETE_STATUS:
-      return JS::BooleanValue(IsDeleteRequested());
+      return AsSomeVariant(IsDeleteRequested());
 
     case LOCAL_GL_COMPILE_STATUS:
-      return JS::BooleanValue(mCompilationSuccessful);
+      return AsSomeVariant(mCompilationSuccessful);
 
     default:
       mContext->ErrorInvalidEnumInfo("getShaderParameter: `pname`", pname);
-      return JS::NullValue();
+      return Nothing();
   }
 }
 
-void WebGLShader::GetShaderSource(nsAString* out) const {
-  out->SetIsVoid(false);
-  *out = mSource;
-}
+nsString WebGLShader::GetShaderSource() const { return mSource; }
 
 void WebGLShader::GetShaderTranslatedSource(nsAString* out) const {
   out->SetIsVoid(false);
@@ -293,11 +290,6 @@
 ////////////////////////////////////////////////////////////////////////////////
 // Boilerplate
 
-JSObject* WebGLShader::WrapObject(JSContext* js,
-                                  JS::Handle<JSObject*> givenProto) {
-  return dom::WebGLShader_Binding::Wrap(js, this, givenProto);
-}
-
 size_t WebGLShader::SizeOfIncludingThis(MallocSizeOf mallocSizeOf) const {
   return mallocSizeOf(this) +
          mSource.SizeOfExcludingThisIfUnshared(mallocSizeOf) +
@@ -314,9 +306,4 @@
   LinkedListElement<WebGLShader>::removeFrom(mContext->mShaders);
 }
 
-NS_IMPL_CYCLE_COLLECTION_WRAPPERCACHE_0(WebGLShader)
-
-NS_IMPL_CYCLE_COLLECTION_ROOT_NATIVE(WebGLShader, AddRef)
-NS_IMPL_CYCLE_COLLECTION_UNROOT_NATIVE(WebGLShader, Release)
-
 }  // namespace mozilla